fe/fe-core/src/main/java/com/starrocks/clone/ColocateTableBalancer.java

```diff
@@ -329,6 +332,12 @@ private void matchGroup() {
                                                     tablet.getId(),
                                                     st);
 
+                                            // If tablet has bad replica, `getColocateHealthStatus()` will also return
+                                            // COLOCATE_MISMATCH, we need to replace the backend which owns the bad
+                                            // replica.
+                                            adjustBackendSetIfReplicaBad(idx, tablet, st, groupId,
```

Comment:
I think that a bad replica causes all the tablet to be migrated is a bit rough. How about just remove the replica on that be and copy a new replica? like tablet FORCE_REDUNDANT state.

Reply:
I think the current handling is ok because bad replica is relatively a rare case, this won't cause too much performance problem. And currently we only have COLOCATE_MISMATCH state to clone a replica for colocate tablet, if we don't change the backend set, new replica cannot be cloned based on the current scheduler logic. Of course we can add extra logic to handle this situation, but this will complicate the scheduler. Taken together, i prefer keeping this pr unchanged.